extensions/blocks/contact-form/deprecated.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import { omit } from 'lodash';
+import { __ } from '@wordpress/i18n';
+import { InnerBlocks } from '@wordpress/block-editor';
+import { createBlock } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import defaultAttributes from './attributes';
+
+const deprecatedAttributes = [
+	'submit_button_text',
+	'has_form_settings_set',
+	'submitButtonText',
+	'backgroundButtonColor',
+	'textButtonColor',
+	'customBackgroundButtonColor',
+	'customTextButtonColor',
+	'submitButtonClasses',
+	'hasFormSettingsSet',
+];
+
+export default [ {
+	attributes: {
+		submit_button_text: {
+			type: 'string',
+			default: __( 'Submit', 'jetpack' ),
+		},
+		has_form_settings_set: {
+			type: 'string',
+			default: null,
+		},
+		submitButtonText: {
+			type: 'string',
+			default: __( 'Submit', 'jetpack' ),
+		},
+		backgroundButtonColor: {
+			type: 'string',
+		},
+		textButtonColor: {
+			type: 'string',
+		},
+		customBackgroundButtonColor: {
+			type: 'string',
+		},
+		customTextButtonColor: {
+			type: 'string',
+		},
+		submitButtonClasses: {
+			type: 'string',
+		},
+		...defaultAttributes,
+	},
+	migrate: ( attributes, innerBlocks ) => {
+		const newAttributes = omit( attributes, deprecatedAttributes );
+
+		const buttonAttributes = {
+			text: attributes.submitButtonText || __( 'Submit', 'jetpack' ),
```

Comment:
I wonder if we are losing the button text when migrating the block from 2 versions behind (I think?).
`submit_button_text` is not mapped to anything now, whereas before it was mapped to `submitButtonText`.

What about `text: attributes.submitButtonText || attributes.submit_button_text || __( 'Submit', 'jetpack' )`?

Reply:
Good catch, thanks.